File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
private:
    void fast(){
        ios_base::sync_with_stdio(false);
        cin.tie(NULL);
        cout.tie(NULL);
    }
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        fast();
        sort(intervals.begin(), intervals.end());
        vector<vector<int>> res;
        for(auto num: intervals){
            if(res.empty() || res.back()[1] < num[0]){
                res.push_back(num);
            }
            else{
                res.back()[1]=max(res.back()[1], num[1]);
            }
        }
        return res;
    }
};

File: 0056-merge-intervals/0056-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort()
        result = []
        i = 0
        while i < len(intervals):
            if len(result) == 0:
                result.append(intervals[i])
            else:
                start = end = 0
                if result[-1][1] >= intervals[i][0]:
                    result[-1][0] = min(result[-1][0], intervals[i][0])
                    result[-1][1] = max(result[-1][1], intervals[i][1])
                else:
                    result.append([intervals[i][0], intervals[i][1]])
            
            i += 1
        
        return result